app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,15 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.find_or_create_by_omniauth(auth_hash)
+    session[:uid] = @uid
```

Comment:
Where is @uid coming from?

Also, the key in the session has to match what you are using in our current_user method. What attribute are we finding by in the current_user method in application controller.